projects/auth0-angular/src/lib/auth.service.ts

```diff
@@ -43,42 +46,60 @@ import { AuthClientConfig } from './auth.config';
 })
 export class AuthService implements OnDestroy {
   private isLoadingSubject$ = new BehaviorSubject(true);
-  private isAuthenticatedSubject$ = new BehaviorSubject(false);
   private errorSubject$ = new ReplaySubject<Error>(1);
+  private refreshState$ = new Subject();
 
   // https://stackoverflow.com/a/41177163
   private ngUnsubscribe$ = new Subject();
-
   /**
    * Emits boolean values indicating the loading state of the SDK.
    */
   readonly isLoading$ = this.isLoadingSubject$.asObservable();
 
   /**
-   * Emits boolean values indicating the authentication state of the user. If `true`, it means a user has authenticated.
-   * This depends on the value of `isLoading$`, so there is no need to manually check the loading state of the SDK.
+   * Trigger used to pull User information from the Auth0Client.
+   * Triggers when an event occurs that needs to retrigger the User Profile information.
+   * Such as: Initially, getAccessTokenSilently, getAccessTokenWithPopup and Logout
    */
-  readonly isAuthenticated$ = this.isLoading$.pipe(
+  private isAuthenticatedTrigger$ = this.isLoading$.pipe(
     filter((loading) => !loading),
     distinctUntilChanged(),
-    concatMap(() => this.isAuthenticatedSubject$)
+    switchMap(() =>
+      // To track the value of isAuthenticated over time, we need to merge:
+      //  - the current value
+      //  - the value whenever refreshState$ emits
+      merge(
+        defer(() => this.auth0Client.isAuthenticated()),
+        this.refreshState$.pipe(
+          mergeMap(() => this.auth0Client.isAuthenticated())
+        )
+      )
+    ),
+    // Ensure every future subscriber receives the last known value
+    shareReplay(1)
```

Comment:
I made some changes to this snippet:

- I used `switchMap` instead of `mergeMap` here because everytime `isLoading$` emits, a new Observable is created  (`merge(...)`). When a subscription is added to the outer observable (`isAuthenticatedTrigger$`), there is also a subscription active on the inner observable (`merge()`).  Whenever isLoading$ emits, a new observable is created without its subscription being unsubscribed from. After `n` emits, we have `n` opwn subscriptions, so that's a memory leak.

We know that isLoading does not emit multiple times, yet it is safer to incorporate that and use a switchMap.

- I merged `defer(() => this.auth0Client.isAuthenticated())` with `this.refreshState$.pipe(mergeMap(() => this.auth0Client.isAuthenticated()))` to represent the isAuthenticated value over time. 
  - `defer(() => this.auth0Client.isAuthenticated())` represent the initial value
  - `this.refreshState$.pipe(mergeMap(() => this.auth0Client.isAuthenticated()))` represents the value after initialization, whenever a refresh is triggered as part of getting a token or logging out.

- I added `shareReplay(1)` to ensure the latest value is re-emitted to future subscribers. It is a bit like how a `ReplaySubject` or `BehaviorSubject` works, but without using a Subject, but applying an operator to an Observable.